### ISSUE TITLE
Frontend Crash when creating Event Definition with unfinished filter

### ISSUE THREAD

**Issue by kmerz** [Graylog2/graylog2-server#8105]
## Expected Behavior
Adding a filter to a Event Definition should not result in a stack trace.

## Current Behavior
![http_method](https://user-images.githubusercontent.com/448763/81703963-9166c900-946d-11ea-9d72-0f5e3c2c37e4.gif)

## Steps to Reproduce (for bugs)
1. Create new Filter and Aggregation Event Definition
2. type `http_method:` into the filter
3. :boom: 

## Stack Trace
```
Uncaught TypeError: Cannot read property 'length' of undefined
    at FilterPreview.renderSearchResult (:8080/464cf0aa-5.js:128)
    at FilterPreview.render (:8080/464cf0aa-5.js:152)
    at finishClassComponent (webpack://__%5Bname%5D/./node_modules/react-dom/cjs/react-dom.development.js?:17160)
    at updateClassComponent (webpack://__%5Bname%5D/./node_modules/react-dom/cjs/react-dom.development.js?:17110)
    at beginWork (webpack://__%5Bname%5D/./node_modules/react-dom/cjs/react-dom.development.js?:18620)
    at HTMLUnknownElement.callCallback (webpack://__%5Bname%5D/./node_modules/react-dom/cjs/react-dom.development.js?:188)
    at Object.invokeGuardedCallbackDev (webpack://__%5Bname%5D/./node_modules/react-dom/cjs/react-dom.development.js?:237)
    at invokeGuardedCallback (webpack://__%5Bname%5D/./node_modules/react-dom/cjs/react-dom.development.js?:292)
    at beginWork$1 (webpack://__%5Bname%5D/./node_modules/react-dom/cjs/react-dom.development.js?:23203)
    at performUnitOfWork (webpack://__%5Bname%5D/./node_modules/react-dom/cjs/react-dom.development.js?:22154)
    at workLoopSync (webpack://__%5Bname%5D/./node_modules/react-dom/cjs/react-dom.development.js?:22130)
    at performSyncWorkOnRoot (webpack://__%5Bname%5D/./node_modules/react-dom/cjs/react-dom.development.js?:21756)
    at eval (webpack://__%5Bname%5D/./node_modules/react-dom/cjs/react-dom.development.js?:11089)
    at unstable_runWithPriority (webpack://__%5Bname%5D/./node_modules/react-dom/node_modules/scheduler/cjs/scheduler.development.js?:653)
    at runWithPriority$1 (webpack://__%5Bname%5D/./node_modules/react-dom/cjs/react-dom.development.js?:11039)
    at flushSyncCallbackQueueImpl (webpack://__%5Bname%5D/./node_modules/react-dom/cjs/react-dom.development.js?:11084)
    at flushSyncCallbackQueue (webpack://__%5Bname%5D/./node_modules/react-dom/cjs/react-dom.development.js?:11072)
    at scheduleUpdateOnFiber (webpack://__%5Bname%5D/./node_modules/react-dom/cjs/react-dom.development.js?:21199)
    at Object.enqueueSetState (webpack://__%5Bname%5D/./node_modules/react-dom/cjs/react-dom.development.js?:12639)
    at ConnectStoresWrapper.Component.setState (webpack://__%5Bname%5D/./node_modules/react/cjs/react.development.js?:471)
    at Store.<anonymous> (app.js:94873)
    at EventEmitter.eventHandler (webpack://__%5Bname%5D/./node_modules/reflux-core/lib/PublisherMethods.js?:44)
    at EventEmitter.emit (webpack://__%5Bname%5D/./node_modules/eventemitter3/index.js?:116)
    at Store.trigger (webpack://__%5Bname%5D/./node_modules/reflux-core/lib/PublisherMethods.js?:113)
    at Store.propagateChanges (builtins.js:52656)
    at builtins.js:52744
    at tryCatcher (builtins.js:24858)
    at Promise._settlePromiseFromHandler (builtins.js:22974)
    at Promise._settlePromise (builtins.js:23031)
    at Promise._settlePromise0 (builtins.js:23076)
    at Promise._settlePromises (builtins.js:23156)
    at _drainQueueStep (builtins.js:19953)
    at _drainQueue (builtins.js:19946)
    at Async._drainQueues (builtins.js:19962)
    at Async.drainQueues (builtins.js:19875)
```

## Your Environment
* Graylog Version: 3.3 development (41a04252d7fc16eec1430cb15d518dd22a3f94e1)

**Comment by edmundoa** [comment]
This looks like the same @linuspahl fixed in https://github.com/Graylog2/graylog2-server/pull/8096, @kmerz could you please try again with an updated master?

**Comment by kmerz** [comment]
> This looks like the same @linuspahl fixed in #8096, @kmerz could you please try again with an updated master?

As you predicted! Thanks @edmundoa
